src/Computer/GlideComputerAirData.cpp

```diff
@@ -134,8 +134,10 @@ GlideComputerAirData::ProcessVertical(const MoreData &basic,
 
   if (calculated.circling)
     CurrentThermal(basic, calculated, calculated.current_thermal);
-  else
+  else {
     calculated.current_thermal = calculated.last_thermal;
+    calculated.current_thermal.Clear();
```

Comment:
Why?

Reply:
But ... why assign `current_thermal` and then clear everything of it? This way, the assignment has no effect. This doesn't appear to make sense.

Reply:
Hm, true, it's not a reference.

Reply:
Removed the assignment, completed the Clear().

Reply:
And why does it need to be cleared?

Reply:
"loss" is a sum that is expected to start from beginning of circling. It needs to be cleared when switching to circling mode. The current implementation of this callback does allow to distiguish circling mode from none circling mode. Thus we can clear, whenever not in circling mode to have the sum zeroed from entering circling mode on.
I am open for more efficiant ways here. The choosen one should be bullet proof.

Reply:
I don't understand your response. It begins with explaining something about "loss", but this part of the change is not about "loss", it's about clearing `calculated.current_thermal`, but none of what you just wrote convinces me that `calculated.current_thermal` should be cleared.
If not circling, `calculated.current_thermal` should contain information about the most recent thermal; it is explicitly implemented that way; clearing it would make that information unavailable.

Reply:
Ok, my understanding is that with setting the end of thermal time stamp and changing the circling mode the TC struct is copied over to the "last TC" variable and the TC is unused until you hit the next thermal. I'll look deeper into that.